spot_driver/spot_driver/ros_helpers.py

```diff
@@ -556,6 +560,16 @@ def get_battery_states_from_state(state: robot_state_pb2.RobotState, spot_wrappe
         battery_msg.status = battery.status
         battery_states_array_msg.battery_states.append(battery_msg)
 
+        if battery_msg.charge_percentage <= 10 and not node.get_parameter("low_battery").value:
+            low_battery_param = Parameter("low_battery", Parameter.Type.BOOL, True)
+            node.set_parameters([low_battery_param])
+            messagebox.showwarning(
+                title="Warning: Low Battery {}".format(battery_msg.identifier),
+                message=(
+                    "Battery is at {} %. Approximately {} minutes remaining.\n Please charge your Spot soon."
+                ).format(battery_msg.charge_percentage, round(battery_msg.estimated_runtime.sec / 60)),
+            )
```

Comment:
This is separate from my main comment, but it would be better to put this user-facing message higher up in the code, for example at the same level where `get_battery_states_from_state` is called in the `spot_ros2` node.

The `get_battery_states_from_state` function is a pretty low-level conversion function that's just responsible for converting a protobuf message to a ROS message, so it shouldn't pop up a UI element.

Reply:
ahhhh i forgot about the driver and driver_cpp merge. I think you're right and I should make a new node